### ISSUE TITLE
Callbacks not working

### ISSUE THREAD

**Issue by francisjervis** [onfido/onfido-sdk-ui#25]
Hi, with the following code I can successfully initialize the document capture modal. However, none of functions that should be called when the capture  (or other event) completes is called:

    Onfido.init({
      token: '{{onfidojwt}}',
      buttonId: 'onfido-button',
      containerId: 'onfido-mount',

      onReady: function() {
        // callback that fires when successfully authorised
        console.log('ready')
      },
      
      onDocumentCapture: function() {
          console.log('capture')
        // callback for when the document has captured successfully
      },
      onFaceCapture: function() {
            console.log('facecapture')
    
        // callback for when the face capture was successful
      },
      onComplete: function() {
        // callback for when everything is complete.
              console.log('complete')
              
              var data = Onfido.getCaptures()
            console.log(data)
    
    
      }
    })

**Comment by hjheath** [comment]
Can you share any console logs you have from the page where you initialise the SDK?

Are you successfully setting up the websockets connection (you will get error logs if this fails, and the document capture will not work)?

**Comment by gosseti** [comment]
@francisjervis I have updated the SDK with changes in place that should get the callbacks working properly. Could you update on your end and let me know if they’re now working?

**Comment by francisjervis** [comment]
Thanks for looking at this. Unfortunately it looks like the master sdk-ui branch is now showing content from the demo. It is also now not detecting the camera at all & does not request access (see #26).

NB also that the screen shot attached is from Chrome at desktop resolution; in responsive mode the “open your bank account” UI disappears! 

Sent from
https://polymail.io/

On Tue, Jun 21, 2016 at 10:52 AM Lawrence Gosset

<
mailto:Lawrence Gosset <notifications@github.com>
> wrote:

https://github.com/francisjervis
I have updated the SDK with changes in place that should get the callbacks working properly. Could you update on your end and let me know if they’re now working?

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/onfido/onfido-sdk-ui/issues/25#issuecomment-227518928
, or
https://github.com/notifications/unsubscribe/AKCFNhg4gRQSVXIz_7qbas9MJfd30lILks5qOCTggaJpZM4I353p
.

https://github.com/onfido/onfido-sdk-ui/issues/25#issuecomment-227518928

**Comment by francisjervis** [comment]
Looking at the code I am guessing this is an issue with /Welcome/index.js being imported wrongly, since that is the only place where the UI element ie.

    <div class='onfido-mtop-large onfido-mbottom-large'>

could be referenced from.

**Comment by rfreitas** [comment]
Hi Francis,

Still having issues? Some issues have been resolved that might be related with what you talked about.

**Comment by francisjervis** [comment]
No, still the same.

**Comment by rfreitas** [comment]
Can you run `npm run dev` on your local repo and go to https://0.0.0.0:8080/
and let me know what you get on your machine?

**Comment by francisjervis** [comment]
I'm not using npm, sorry.

**Comment by rfreitas** [comment]
then how are you building it? The dependencies are all managed by npm.

**Comment by francisjervis** [comment]
The docs say "Include it as a regular script tag on your page" *or* use npm. So I am including this source at the head of the page:
https://cdn.rawgit.com/onfido/onfido-sdk-ui/master/dist/onfido.min.js

**Comment by rfreitas** [comment]
I see. The dist folder seems to be out of sync with the code.

Please try this onfido script for now: https://rawgit.com/onfido/onfido-sdk-ui/24a264a70163c000e371738d818539a675df32c1/dist/onfido.min.js

Here is a Jsfiddle running it:
https://jsfiddle.net/4xqtt6fL/

**Comment by francisjervis** [comment]
Thanks. I am getting this error after uploading a face image though: "onfido.min.js:6 Uncaught TypeError: e.toDataURL is not a function" - hangs on the screen shown. It's still not showing anything on an actual mobile size screen.
![screen shot 2016-07-25 at 2 30 53 am](https://cloud.githubusercontent.com/assets/10519862/17096819/0ea38826-5210-11e6-829d-9963daf2de74.png)

**Comment by rfreitas** [comment]
Hey Francis,

I was able to go through all the steps on JsFiddle. Can you confirm you have the same issue on the JsFiddle?

**Comment by francisjervis** [comment]
OK I see what is happening. The updated JS (I also had to add the CSS stylesheet tag using the rawgit link from the jsfiddle) fires the callbacks, but when I try to get the results in the onComplete block, get this:

"VM736 onfido.min.js:6 Uncaught ReferenceError: captureSelector is not defined"

This also happens in JSFiddle https://jsfiddle.net/k25mcntn/

**Comment by rfreitas** [comment]
Alright, great news. We caught the bug.

Here is the stacktrace:
```
(index):91 Uncaught ReferenceError: captureSelector is not defined
p.getCaptures @ onfido.min.js:6
m.getCaptures @ onfido.min.js:1
onComplete @ (index):91
(anonymous function) @ onfido.min.js:1
r @ onfido.min.js:7s.emit @ onfido.min.js:7
i @ onfido.min.js:6
l @ onfido.min.js:6
dispatch @ VM195:1570(anonymous function) @ onfido.min.js:8
face @ onfido.min.js:6
o.handleImage @ onfido.min.js:6
(anonymous function) @ onfido.min.js:6
r @ onfido.min.js:7s.emit @ onfido.min.js:7
(anonymous function) @ onfido.min.js:6
a.onload @ onfido.min.js:7
```

**Comment by rfreitas** [comment]
Hey Francis,

Can you try the latest code? I was able to fix it on my config.

**Comment by rfreitas** [comment]
Fixed on 0.7.0
